pkg/store/proxy.go

```diff
@@ -255,305 +270,209 @@ func (s *ProxyStore) Series(r *storepb.SeriesRequest, srv storepb.Store_SeriesSe
 				"store.addr": st.Addr(),
 			})
-
-			sc, err := st.Series(seriesCtx, r)
+			cl, err := st.Series(seriesCtx, actualRequest)
 			if err != nil {
 				err = errors.Wrapf(err, "fetch series for %s %s", storeID, st)
+				errMtx.Lock()
+				errs.Add(err)
+				errMtx.Unlock()
+
 				span.SetTag("err", err.Error())
 				span.Finish()
-				if r.PartialResponseDisabled {
-					level.Error(reqLogger).Log("err", err, "msg", "partial response disabled; aborting request")
-					return err
-				}
-				respSender.send(storepb.NewWarnSeriesResponse(err))
-				continue
+				return
 			}
 
-			// Schedule streamSeriesSet that translates gRPC streamed response
-			// into seriesSet (if series) or respCh if warnings.
-			seriesSet = append(seriesSet, startStreamSeriesSet(seriesCtx, reqLogger, span, closeSeries,
-				wg, sc, respSender, st.String(), !r.PartialResponseDisabled, s.responseTimeout, s.metrics.emptyStreamResponses))
-		}
-
-		level.Debug(reqLogger).Log("msg", "Series: started fanout streams", "status", strings.Join(storeDebugMsgs, ";"))
-
-		if len(seriesSet) == 0 {
-			// This is indicates that configured StoreAPIs are not the ones end user expects.
-			err := errors.New("No StoreAPIs matched for this query")
-			level.Warn(reqLogger).Log("err", err, "stores", strings.Join(storeDebugMsgs, ";"))
-			respSender.send(storepb.NewWarnSeriesResponse(err))
-			return nil
-		}
-
-		// TODO(bwplotka): Currently we stream into big frames. Consider ensuring 1MB maximum.
-		// This however does not matter much when used with QueryAPI. Matters for federated Queries a lot.
-		// https://github.com/thanos-io/thanos/issues/2332
-		// Series are not necessarily merged across themselves.
-		mergedSet := storepb.MergeSeriesSets(seriesSet...)
-		for mergedSet.Next() {
-			lset, chk := mergedSet.At()
-			respSender.send(storepb.NewSeriesResponse(&storepb.Series{Labels: labelpb.ZLabelsFromPromLabels(lset), Chunks: chk}))
-		}
-		return mergedSet.Err()
-	})
-	g.Go(func() error {
-		// Go routine for gathering merged responses and sending them over to client. It stops when
-		// respCh channel is closed OR on error from client.
-		for resp := range respCh {
-			if err := srv.Send(resp); err != nil {
-				return status.Error(codes.Unknown, errors.Wrap(err, "send series response").Error())
-			}
-		}
-		return nil
-	})
-	if err := g.Wait(); err != nil {
-		// TODO(bwplotka): Replace with request logger.
-		level.Error(reqLogger).Log("err", err)
-		return err
-	}
-	return nil
-}
-
-type directSender interface {
-	send(*storepb.SeriesResponse)
-}
-
-// streamSeriesSet iterates over incoming stream of series.
-// All errors are sent out of band via warning channel.
-type streamSeriesSet struct {
-	ctx    context.Context
-	logger log.Logger
-
-	stream storepb.Store_SeriesClient
-	warnCh directSender
+			rCh := make(chan *recvResponse)
+			done := make(chan struct{})
+			go func() {
+				for {
+					// TODO: we can buffer this in files to have infinitely
```

Comment:
```suggestion
					// TODO(GiedriusS): we can buffer this in files to have infinitely
```

Please add your username if you authored TODO, otherwise we don't know how might have more context about.